### PR TITLE
KAFKA-8538 (part of KIP-345): add group.instance.id to DescribeGroup

### PR DESCRIPTION
**Pull request by abbccdda**
Include group.instance.id in the describe group result for better visibility.
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala
Comment (abbccdda) [inline core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala]: @hachikuji @guozhangwang I noticed that this class was only used in the deprecated admin client, do we still need to add `group.instance.id` support to it? If not, I could consider removing the follow-up changes in scala code.

**Comment by abbccdda** [comment]
retest this please

File: core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala
Comment (guozhangwang) [reply core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala]: I think @ijuma already have a PR for removing the deprecated admin clients. We can wait for his PR to be merged first?

File: clients/src/main/java/org/apache/kafka/clients/admin/MemberDescription.java
Comment (guozhangwang) [inline clients/src/main/java/org/apache/kafka/clients/admin/MemberDescription.java]: Actually the constructor of all such description classes can just be default package-private since they are only used by KafkaAdminClient, and hence can just be private APIs, and we do not need to deprecate-overload any more.

File: core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala
Comment (ijuma) [reply core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala]: I merged it.

File: core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala
Comment (ijuma) [inline core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala]: How come we're replacing instead of adding a new test?

Comment (abbccdda) [reply core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala]: Because we only need to replace a dynamic member with a static member so that we could retrieve a valid `group.instance.id` for testing purpose. Should be sufficient because all the other test outputs will be expected to be same.

Comment (ijuma) [reply core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala]: Yeah, but we want to test this scenario with both static and dynamic, right?

Comment (abbccdda) [reply core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala]: Sounds good to me

File: clients/src/main/java/org/apache/kafka/clients/admin/MemberDescription.java
Comment (guozhangwang) [inline clients/src/main/java/org/apache/kafka/clients/admin/MemberDescription.java]: As mentioned above, we can make this constructor default access.

**Review by guozhangwang — COMMENTED** [review]
LGTM except one minor comment, ping another pair of eyes before merging.

**Comment by abbccdda** [comment]
Retest this please

**Comment by abbccdda** [comment]
Retest this please

**Comment by abbccdda** [comment]
@ableegoldman @vvcephei Could you also take a look?

**Review by ableegoldman — APPROVED** [review]
LGTM

**Comment by abbccdda** [comment]
@ableegoldman Thank you!

**Comment by abbccdda** [comment]
Retest this please

**Pull request merged**
